contrib/mpi-proxy-split/lower-half/gethostbyname-static/gethostbyname_static.c

```diff
@@ -75,11 +81,26 @@ static int error(struct hostent **result) {
 }
 
 
-char *getProxyPath(char *buf, size_t buflen) {
-  readlink("/proc/self/exe", buf, buflen);
-  char *dir = dirname(buf);
-  assert(strlen(dir) + sizeof("gethostbyname_proxy") < buflen);
-  return strcat(dir, "/gethostbyname_proxy");
+// For getaddrinfo or gethostbyname_r
+void execvpProxyWithArgv(const char *file, const char *arg1, const char *arg2) {
+    char *argv[] = {(char *)file, (char *)arg1, (char *)arg2, NULL};
+    int rc = execvp("execvp(gethostbyname_proxy)", argv);
```

Comment:
I just realized that the `file` passed here, is actually `gethostbyname_r` or `getaddrinfo`, which is the argument that will be passed to `gethostbyname_proxy`.
But by convention, I think the first element in the `argv` should be `gethostbyname_proxy`?
> The execv(), execvp(), and execvpe() functions provide an array of pointers to null-terminated strings that represent the argument list available to the new program. The first argument, by convention, should point to the filename associated with the file being executed. The array of pointers must be terminated by a NULL pointer.

And I guess there is a typo? `execvp("gethostbyname_proxy", argv);`

Reply:
i did some testing, and jun is right. doing `execvp("execvp(gethostbyname_proxy)", argv)` causes `execvp` to fail.

Reply:
Hi @fyshhh and @jungan ,
    As usual, you're right again.  It was a silly error on my part.  I've updated the PR to fix this.  I believe the execvp call works correctly now.

Reply:
great! did some testing on it and seems like it's working properly now.